pkg/file/builder.go

```diff
@@ -1755,20 +1755,50 @@ func (b *stateBuilder) fillPluginConfig(plugin *FPlugin) error {
 	return nil
 }
 
-func pluginRelations(plugin *kong.Plugin) (cID, rID, sID, cgID string) {
+func (b *stateBuilder) pluginRelations(plugin *kong.Plugin) (cID, rID, sID, cgID string) {
 	if plugin.Consumer != nil && !utils.Empty(plugin.Consumer.ID) {
-		cID = *plugin.Consumer.ID
+		consumer, err := b.currentState.Consumers.GetByIDOrUsername(*plugin.Consumer.ID)
```

Comment:
We're making this change so actual ID is returned instead of name when foreign key reference exists, correct?

Reply:
Yes!
The internal state that we maintain uses IDs. So, if a name is passed to it, even if the said entity exists, the lookup happens via the name and fails with a 404.

Reply:
https://github.com/Kong/go-database-reconciler/blob/63d97250ecf59707d722d0646b1e0d6912bc9d00/pkg/file/builder.go#L1711 is the lookup that needs the ID - and returns nil if name is given - leading to unique violation eventually as we try to create instead of update - correct?

Reply:
Yes. Correct!